dist/js/bootstrap-multiselect.js

```diff
@@ -1231,6 +1231,57 @@
             this.rebuild();
         },
 
+        dataproviderOnlyReplaceOptions: function(dataprovider) {
+            var groupCounter = 0;
+            var $select = this.$select.empty();
+
+            $.each(dataprovider, function (index, option) {
+                var $tag;
+
+                if ($.isArray(option.children)) { // create optiongroup tag
+                    groupCounter++;
+
+                    $tag = $('<optgroup/>').attr({
+                        label: option.label || 'Group ' + groupCounter,
+                        disabled: !!option.disabled
+                    });
+
+                    forEach(option.children, function(subOption) { // add children option tags
+                        $tag.append($('<option/>').attr({
+                            value: subOption.value,
+                            label: subOption.label || subOption.value,
+                            title: subOption.title,
+                            selected: !!subOption.selected,
+                            disabled: !!subOption.disabled
+                        }));
+                    });
+                }
+                else {
+                    $tag = $('<option/>').attr({
+                        value: option.value,
+                        label: option.label || option.value,
+                        title: option.title,
+                        selected: !!option.selected,
+                        disabled: !!option.disabled
+                    });
+                }
+
+                $select.append($tag);
+            });
+
+            var allDropdownOptions = $("li:not(.multiselect-item)", this.$ul);
```

Comment:
maybe extracting a simple factory function that generates a data provider function with a hook would be better then.. want to pair on it?